quota/etcd/quotaapi/conversions.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package quotaapi
+
+import (
+	"fmt"
+
+	"github.com/google/trillian/quota/etcd/quotapb"
+	"github.com/google/trillian/quota/etcd/storagepb"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+const (
+	statePath           = "state"
+	maxTokensPath       = "max_tokens"
+	sequencingBasedPath = "sequencing_based"
+	timeBasedPath       = "time_based"
+)
+
+var (
+	validPaths = map[string]bool{
+		statePath:           true,
+		maxTokensPath:       true,
+		sequencingBasedPath: true,
+		timeBasedPath:       true,
+	}
+
+	commonMask          = &field_mask.FieldMask{Paths: []string{statePath, maxTokensPath}}
+	sequencingBasedMask = &field_mask.FieldMask{Paths: append(commonMask.Paths, sequencingBasedPath)}
+	timeBasedMask       = &field_mask.FieldMask{Paths: append(commonMask.Paths, timeBasedPath)}
+
+	errBothReplenishmentStrategies = fmt.Errorf("both %q and %q paths specified, please pick a single replenishment strategy", sequencingBasedPath, timeBasedPath)
+)
+
+// validateMask returns nil if mask is a valid Config mask, error otherwise.
+// Paths must match the proto name of the fields (e.g., "time_based", not "TimeBased").
+// If fields belong to a oneof (such as "sequencing_based" and "time_based"), then only one field of
+// the oneof may be specified.
+func validateMask(mask *field_mask.FieldMask) error {
+	sequencingBasedFound := false
+	timeBasedFound := false
+	for _, path := range mask.Paths {
+		switch {
```

Comment:
Given this is switching on `path`, it would be easier to follow as a `switch path`, dropping `validPaths` and adding:
```go
	case statePath, maxTokensPath:
		// ok
	default:
￼		return fmt.Errorf("invalid field path for Config: %q", path)
```